recipes/hdf5/all/conanfile.py

```diff
@@ -39,123 +43,119 @@ class Hdf5Conan(ConanFile):
         "parallel": False,
     }
 
-    generators = "cmake"
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
-
     def export_sources(self):
-        self.copy("CMakeLists.txt")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
+    def validate(self):
+        if not can_run(self):
+            # While building it runs some executables like H5detect
+            raise ConanInvalidConfiguration("Current recipe doesn't support cross-building (yet)")
+        if self.info.options.parallel:
+            if self.info.options.enable_cxx:
+                raise ConanInvalidConfiguration("Parallel and C++ options are mutually exclusive")
+            if self.info.options.get_safe("threadsafe"): # FIXME why can't I define the default valid as False?
```

Comment:
What exactly is the issue?

Reply:
This is apparently some built-in limitation with HDF5.
Parallel == MPI so its not just about enabling threads.
I think only Linux and OpenMPI is possible at the moment too.